classes/controller/FrontController.php

```diff
@@ -191,6 +191,9 @@ class FrontControllerCore extends Controller
      */
     protected $automaticallyAllocateDeliveryAddress = true;
 
+    /** @var string Page name */
```

Comment:
```suggestion
    /** @var string|null Page name */
```

Reply:
It's not defined as null, it's void here. this is why it's not `|null` :thinking:

Reply:
If it's not null, in this case, the property must have a default value (`''`).

Reply:
It's void, this is why it's not defined. Otherwise, PHPStan would have screamed a lot, no?

Reply:
<img src='https://media.giphy.com/media/pUeXcg80cO8I8/source.gif'>

Reply:
```
    /** @var string Page name */
   public $page_name = '';
```
or
```
    /** @var string|null Page name */
   public $page_name;
```

Reply:
Normally if we put string and don't define a default value, phpstan should yell at us

Reply:
I confirm that PHPStan will not trigger anything
If you have
```php
/**
 * My method
 */
public function myMethod()
{
}
```
You're not adding `@return null` because it's the same as 

```php
/**
 * My method
 */
public function myMethod():void
{
}
```


As you can see in my screenshot:
![image](https://user-images.githubusercontent.com/1462701/102897049-afff5a80-4467-11eb-8d5c-53c89758a93b.png)

Everything is working well and no errors are trigger by phpstan even in level 8.
PHPStan will check if something is wrong when you're trying to set a new value in the property, otherwise as I said earlier, it's like many other languages, it's void.
Think about procedure and function which are not the same :wink:

Reply:
Forget something.

If you add the `null` this means you allow anyone to do `$this->page_name = null` after having set the `page_name` :thinking:

Reply:
@PululuK you see what you did!! 😂